main_test.go

```diff
@@ -72,3 +75,64 @@ func (t *MainTest) TestGetUserAgentWhenMetadataImageTypeEnvVarSetAndAppNameNotSe
 
 	ExpectEq(expectedUserAgent, userAgent)
 }
+
+func (t *MainTest) TestStringifyShouldReturnAllFlagsPassedInMountConfigAsMarshalledString() {
+	mountConfig := &config.MountConfig{
+		WriteConfig: config.WriteConfig{
+			CreateEmptyFile: false,
+		},
+		LogConfig: config.LogConfig{
+			Severity: config.TRACE,
+			FilePath: "\"path\"to\"file\"",
+			LogRotateConfig: config.LogRotateConfig{
+				MaxFileSizeMB:   2,
+				BackupFileCount: 2,
+				Compress:        true,
+			},
+		},
+	}
+
+	actual := stringify(mountConfig)
+
+	expected := "{\"CreateEmptyFile\":false,\"Severity\":\"TRACE\",\"Format\":\"\",\"FilePath\":\"\\\"path\\\"to\\\"file\\\"\",\"LogRotateConfig\":{\"MaxFileSizeMB\":2,\"BackupFileCount\":2,\"Compress\":true}}"
+	AssertEq(strings.TrimSpace(expected), strings.TrimSpace(actual))
+}
+
+func (t *MainTest) TestStringifyShouldReturnAllFlagsPassedInFlagStorageAsMarshalledString() {
+	mountOptions := map[string]string{
+		"1": "one",
+		"2": "two",
+		"3": "three",
+	}
+	flags := &flagStorage{
+		SequentialReadSizeMb: 10,
+		ClientProtocol:       mountpkg.ClientProtocol("http4"),
+		MountOptions:         mountOptions,
+	}
+
+	actual := stringify(flags)
+
+	expected := "{\"AppName\":\"\",\"Foreground\":false,\"ConfigFile\":\"\",\"MountOptions\":{\"1\":\"one\",\"2\":\"two\",\"3\":\"three\"},\"DirMode\":0,\"FileMode\":0,\"Uid\":0,\"Gid\":0,\"ImplicitDirs\":false,\"OnlyDir\":\"\",\"RenameDirLimit\":0,\"CustomEndpoint\":null,\"BillingProject\":\"\",\"KeyFile\":\"\",\"TokenUrl\":\"\",\"ReuseTokenFromUrl\":false,\"EgressBandwidthLimitBytesPerSecond\":0,\"OpRateLimitHz\":0,\"SequentialReadSizeMb\":10,\"MaxRetrySleep\":0,\"StatCacheCapacity\":0,\"StatCacheTTL\":0,\"TypeCacheTTL\":0,\"HttpClientTimeout\":0,\"MaxRetryDuration\":0,\"RetryMultiplier\":0,\"LocalFileCache\":false,\"TempDir\":\"\",\"ClientProtocol\":\"http4\",\"MaxConnsPerHost\":0,\"MaxIdleConnsPerHost\":0,\"EnableNonexistentTypeCache\":false,\"StackdriverExportInterval\":0,\"OtelCollectorAddress\":\"\",\"LogFile\":\"\",\"LogFormat\":\"\",\"ExperimentalEnableJsonRead\":false,\"DebugFuseErrors\":false,\"DebugFuse\":false,\"DebugFS\":false,\"DebugGCS\":false,\"DebugHTTP\":false,\"DebugInvariants\":false,\"DebugMutex\":false}"
+	AssertEq(strings.TrimSpace(expected), strings.TrimSpace(actual))
+}
+
+func (t *MainTest) TestStringifyShouldReturnEmptyStringWhenMarshalErrorsOut() {
+
+	customInstance := CustomType{
+		Value: "example",
+	}
+
+	actual := stringify(customInstance)
+
+	expected := ""
+	AssertEq(strings.TrimSpace(expected), strings.TrimSpace(actual))
+}
+
+type CustomType struct {
```

Comment:
nit: Is it mandatory to make this object public? Can't we keep `customType`?

Reply:
made it private